### ISSUE TITLE
Deploy Ruby on Rails with Epinio

### ISSUE THREAD

**Issue by jimmykarily** [epinio/epinio#441]
Depends on #375 

Deploying a Rails application requires that we set the RAILS_MASTER_KEY environment variable during build. This should be possible with a project.toml like this:

```
~/workspace/epinio (main)*$ cat rails/epinio-sample/project.toml 
[[build.env]]
RAILS_MASTER_KEY="key_goes_here"
```
but didn't seem to work on the first try.

Also consider enabling the CACHE_IMAGE to make builds faster:

The following results in an error:

```
--- a/assets/embedded-files/tekton/pipeline.yaml
+++ b/assets/embedded-files/tekton/pipeline.yaml
@@ -52,6 +52,8 @@ spec:
       value: paketobuildpacks/builder:full
     - name: SOURCE_SUBPATH
       value: app
+    - name: CACHE_IMAGE
+      value: "$(params.IMAGE)-cache"

```

```
🕞  [cac8b614852bae8f-stage-vqqwl-pod-bs2sf] step-create ERROR: failed to create image cache: accessing cache image "127.0.0.1:30500/apps/rails-test-cb92b559e6b3a2f50718b142a00682a24a6eee9c-cache": connect to repo store "127.0.0.1:30500/apps/rails-test-cb92b559e6b3a2f50718b142a00682a24a6eee9c-cache": Get "https://127.0.0.1:30500/v2/": dial tcp 127.0.0.1:30500: connect: connection refused; Get "http://127.0.0.1:30500/v2/": dial tcp 127.0.0.1:30500: connect: connection refused 
```

## Acceptance Criteria

- A vanilla Rails application can be deployed with Epinio
- It should be possible to use a Mariadb database (e.g. with the incluster services)
- [Optional] Try to deploy some popular Rails application to discover more possible issues on Epinio.
- Findings should be documented in a README of a sample app with the all the needed code available. That repository should be used as a reference for people deploying Rails apps in the future.

**Comment by jimmykarily** [comment]
An example on how to run a PipelineRun with env vars: https://github.com/tektoncd/catalog/blob/45742091e57f996e4ea758bc7404f59a4f04e952/task/buildpacks-phases/0.2/README.md#usage

**Comment by jimmykarily** [comment]
I pushed all the changes I had to make to be able to stage Rails here: https://github.com/epinio/epinio/tree/rails_fixes
(it has a hardcoded environment variable used to check if this fixes staging).
Proper environment variable support will be implemented on #375 which is now a dependency for this ticket.
The tmp branch also shows how to use the CACHE_IMAGE feature to speed up staging. It makes a huge difference for apps like Rails which has to run `bundle install` and doing it without caches could take 5 minutes or even more.

Proper CACHE_IMAGE support should also be handled on another ticket. It involves keeping track of the cache images, cleaning up when the application is deleted, deleting the cache on use demand and maybe pushing an application ignoring caches (when cache creates some problem for whatever reason).

**Comment by jimmykarily** [comment]
The experiments where done using this application: https://github.com/epinio/example-rails

The README of that repository should be updated when we are done to properly document how to run a Rails application on Epinio.

**Comment by jimmykarily** [comment]
We updated the instructions to deploy Rails: https://github.com/epinio/example-rails/pull/1
We only need to add an acceptance test in Epinio to push Rails and we can then close this issue.

Ideally Epinio would provide  way to run one-off commands for things like migrations. Right now we worked around the issue by creating a Procfile that prepares the database before starting the server: https://github.com/epinio/example-rails/pull/1/files#diff-0a99231995da379e7aebabe76c9d849a23737a42c3b3a8994043e2aa80958424R1
